libs/community/langchain_community/chat_models/writer.py

```diff
@@ -113,24 +54,26 @@ class ChatWriter(BaseChatModel):
         .. code-block:: python
 
             from langchain_community.chat_models import ChatWriter
+            from writerai import Writer, AsyncWriter
+
+            client = Writer()
+            async_client = AsyncWriter()
 
-            chat = ChatWriter(model="palmyra-x-004")
+            chat = ChatWriter(
+                client=client,
+                async_client=async_client,
```

Comment:
jfyi this is a bit of an antipattern. you're welcome to do this if you prefer

Reply:
would recommend initializing client/async client in a model validator

Reply:
That's what I wanted to do, but in case of such an implementation I need to import "writer-sdk" external dependency. Locally I can do that just by adding to poetry deps, but in case of it at the stage of running GitHub actions error with checking imports appears and also tests can't be collected due to miss of imported package. I read [docs](https://python.langchain.com/docs/contributing/how_to/code/setup/#working-with-optional-dependencies) about optional deps for testing, add "writer-sdk" to requirements for testing and tried various approaches of "required" usage but didn't achieve any success. I think the problem that I try to import optional package not even in tests but in code of the wrapper. If you can help me to pass imports checking and tests collecting in case of external import in wrapper file, it will be possible to implement model creation that satisfies your request. Waiting for your reply, thank you!

Reply:
hey! would recommend looking at other chat model implementations to see how they do it. here's our guide on using optional imports: https://python.langchain.com/docs/contributing/how_to/code/setup/#working-with-optional-dependencies